frame/collective/src/dispatch_account_tests.rs

```diff
@@ -0,0 +1,292 @@
+		frame_system::limits::BlockWeights::simple_max(1024);
+}
+impl frame_system::Config for Test {
+	type BaseCallFilter = ();
+	type BlockWeights = ();
+	type BlockLength = ();
+	type DbWeight = ();
+	type Origin = Origin;
+	type Index = u64;
+	type BlockNumber = u64;
+	type Call = Call;
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type AccountId = AccountId;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	type Event = Event;
+	type BlockHashCount = BlockHashCount;
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = ();
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type SS58Prefix = ();
+	type OnSetCode = ();
+}
+impl Config<Instance1> for Test {
+	type Origin = Origin;
+	type Proposal = Call;
+	type Event = Event;
+	type MotionDuration = MotionDuration;
+	type MaxProposals = MaxProposals;
+	type MaxMembers = MaxMembers;
+	type DefaultVote = PrimeDefaultVote;
+	type ModuleId = ModuleId1;
+	type WeightInfo = ();
+}
+impl Config<Instance2> for Test {
+	type Origin = Origin;
+	type Proposal = Call;
+	type Event = Event;
+	type MotionDuration = MotionDuration;
+	type MaxProposals = MaxProposals;
+	type MaxMembers = MaxMembers;
+	type DefaultVote = MoreThanMajorityThenPrimeDefaultVote;
+	type ModuleId = ModuleId2;
+	type WeightInfo = ();
+}
+impl Config for Test {
+	type Origin = Origin;
+	type Proposal = Call;
+	type Event = Event;
+	type MotionDuration = MotionDuration;
+	type MaxProposals = MaxProposals;
+	type MaxMembers = MaxMembers;
+	type DefaultVote = PrimeDefaultVote;
+	type ModuleId = ModuleId0;
+	type WeightInfo = ();
+}
+
+// example module to test behaviors.
+pub mod example {
+	use super::*;
+	use frame_system::ensure_signed;
+	pub trait Config: frame_system::Config { }
+
+	decl_storage! {
+		trait Store for Module<T: Config> as Example {
+			pub WhoCalled: Vec<T::AccountId>;
+		}
+	}
+
+	decl_module! {
+		pub struct Module<T: Config> for enum Call where origin: <T as frame_system::Config>::Origin {
+			#[weight = 0]
+			fn store_me(
+				origin,
+			) -> DispatchResult {
+				let who = ensure_signed(origin)?;
+				WhoCalled::<T>::append(who);
+				Ok(())
+			}
+		}
+	}
+}
+
+impl example::Config for Test {}
+
+pub type Block = sp_runtime::generic::Block<Header, UncheckedExtrinsic>;
+pub type UncheckedExtrinsic = sp_runtime::generic::UncheckedExtrinsic<u32, u64, Call, ()>;
+
+frame_support::construct_runtime!(
+	pub enum Test where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic
+	{
+		System: frame_system::{Pallet, Call, Event<T>},
+		Collective: collective::<Instance1>::{Pallet, Call, Event<T>, Origin<T>, Config<T>},
+		CollectiveMajority: collective::<Instance2>::{Pallet, Call, Event<T>, Origin<T>, Config<T>},
+		DefaultCollective: collective::{Pallet, Call, Event<T>, Origin<T>, Config<T>},
+		Example: example::{Pallet, Call},
+	}
+);
+
+fn account(n: u32) -> AccountId {
```

Comment:
Maybe we could squeeze this into the main tests as well, with a 
```
enum Account {
     Fake(u64), 
     Real(AccountId32),
}
```

Not worth the effort I guess